viam-cartographer/src/slam_service/slam_service.h

```diff
@@ -252,6 +256,9 @@ class SLAMServiceImpl final : public SLAMService::Service {
     // started.
     std::string latest_pointcloud_map;
     // ---
+
+    cartographer::mapping::TrajectoryBuilderInterface *trajectory_builder;
+    int trajectory_id;
```

Comment:
@kkufieta , do we need to access `trajectory_builder` and `trajectory_id` under `map_builder_mutex`?

If so @kim-mishra could we move this to just after L244, so that it's grouped with things that need to be accessed under `map_builder_mutex`?

Reply:
Yes, `trajectory_builder` should be protected by `map_builder_mutex`. It makes sense to do the same for `trajectory_id` even though it's probably not strictly necessary.

Reply:
Can we please move this to CPU bound initialization time (i.e. point 5)?